### PR TITLE
fixes #1304 Consul ServiceLookupThread connect to consul server get s…

### LINKED ISSUE #1304: Consul ServiceLookupThread connect to consul server get stuck
We experienced connection problem from ConsulRegistry to consul server node so that local consul registry cache could not be synchronised when service registry got updated on server side. We found this issue in one of our `light-router` instances but the implementation logic is in `light-4j`'s Consul module. This issue cannot recover by itself until restart the `light-router` instance.

We know that there is a background ServiceLookupThread that uses blocking query to consul server to get any service registration update. There is a known issue on the consul server (v 1.4.4) side that causes existing connection to consul becomes unusable and no new connection can be created to the server either. There is no report about whether the same issue could happen in new consul versions yet.

In current consul registry implementation, it waits forever for a new connection to be created to consul server without any timeout set. So if there is any problem on the consul side that stop the connection from being created, it could hang the communication from ConsulRegistry to the consul server forever.

This issue exists in both 1.6.x and 2.x branches.

### PR DESCRIPTION
**Pull request by stevehu**
…tuck

### CODE REVIEW

**Review by jsu216 — COMMENTED** [review]
Do we need some logic to reset the connection or pool, if we cannot get any response from consul before timeout?

File: consul/src/main/java/com/networknt/consul/client/ConsulClientImpl.java [hunk 1]
```diff
@@ -246,7 +244,7 @@ AtomicReference<ClientResponse> send(ClientConnection connection, HttpString met
 			request.getRequestHeaders().put(Headers.TRANSFER_ENCODING, "chunked");
 			connection.sendRequest(request, client.createClientCallback(reference, latch, json));
 		}
-		latch.await();
+		latch.await(ConsulUtils.getWaitInSecond(wait), TimeUnit.SECONDS);
 		logger.trace("The response got from consul: {} = {}", uri.toString(), reference.get().toString());
```
[the remark this answers is not in the document]
Comment (stevehu) [reply consul/src/main/java/com/networknt/consul/client/ConsulClientImpl.java]: I have added a trace statement for the timeout condition to clearly identify the situation in the log. The reference can be null and reference.get() will throw an NPE; however, the catch blocks for each caller will just log the error and continue the logic except the registerService which you want to let the server module know the registration is failed and handle it properly during the server startup. 

Based on your suggestion, I have closed the connection if timeout to force the reconnect for the next method call.

**Review by jsu216 — APPROVED** [review]

**Review by miklish — APPROVED** [review]

**Comment by miklish** [comment]
@stevehu pls merge ASAP

**Comment by stevehu** [comment]
A new snapshot version is released. Please test and let me know if it fixes the issue. Thanks.

**Pull request merged**